### ISSUE TITLE
C# TestServer

### ISSUE THREAD

**Issue by texttechne** [odata2ts/odata2ts#161]
@ShaneGH would be cool to take over your TestServer implementation from `magic-odata` and then adapt it depending on the output of #160. I would try to do the same for CAP and Olingo implementations as soon as you have paved the way ;-)

You have full write access to [the repository](https://github.com/odata2ts/test-server-asp-net).
Ideally, at least in my mind, you take over the implementations more or less as-it-is and then adapt as necessary.

**Comment by ShaneGH** [comment]
Sounds good.
@texttechne  Have you done any work on #160 yet, or is this a brand new spec?

**Comment by texttechne** [comment]
No haven't done any work on that not really. Always had it on my mind, but never got around to it.

So yes, it's brand new. And your TestServer is the most advanced example there is 😄.

**Comment by ShaneGH** [comment]
@texttechne 
I think the best thing to do here is to port the server over as is with some documentation. What do you think?

**Comment by texttechne** [comment]
@ShaneGH yes, that woul be the starting point: take everything over. After that I think we should adapt the project depending of the outcome of #160.

**Comment by ShaneGH** [comment]
@texttechne hey, sorry, I have not had a lot of time to spend on this recently.
I have a commit ready to go for https://github.com/odata2ts/test-server-asp-net.git but I seem to have lost access (to main or any other branch)

**Comment by texttechne** [comment]
Hi @ShaneGH,

cool! Nice to hear you're still up for it!

The invitation was expired. Should work now.

**Comment by ShaneGH** [comment]
Ok v1 is done.

- I added a couple of lines into the readme on how to run the server. 
- Once you run it it will print a basic description of itself on the console before starting the web server
- I have added comments to the code rather than in documentation file, but please let me know if there is something more that needs explaining

**Comment by ShaneGH** [comment]
FYI, I pushed a branch before I pushed main, and github has set the default branch to be `aspnet-test-server`
I you might want to set the default back to `main`

**Comment by texttechne** [comment]
Switched the default branch to `main` so that all repos behave the same.

Awesome stuff, thank you so much!
I'll try it out...

**Comment by texttechne** [comment]
See: [test-server-asp-net](https://github.com/odata2ts/test-server-asp-net)
